Adding animations to GLTF export as a postprocessing step?

Hey,

I noticed there's a `tree_postprocessor` argument in the `Scene.export()` method and it got me thinking if it could be possible to add animations to an exported GLTF that way. 

I did some digging and based on the [gltfTutorial_006_SimpleAnimation](https://github.com/KhronosGroup/glTF-Tutorials/blob/master/gltfTutorial/gltfTutorial_006_SimpleAnimation.md) I managed to scrape together something that I think could work:

// polygon_animation.py
```python
scene = trimesh.Scene()

scene.add_geometry(polygon_mesh, node_name="test", geom_name="test")

# https://github.com/KhronosGroup/glTF-Tutorials/blob/master/gltfTutorial/gltfTutorial_006_SimpleAnimation.md
# https://github.com/KhronosGroup/glTF-Tutorials/blob/master/gltfTutorial/gltfTutorial_007_Animations.md
def add_animation_to_tree(tree):
    tree["animations"] = [
        {
            "samplers": [{"input": 2, "interpolation": "LINEAR", "output": 3}],
            "channels": [{"sampler": 0, "target": {"node": 1, "path": "rotation"}}],
        }
    ]

scene.export(file_obj="polygon_animation.glb", file_type=".glb", tree_postprocessor=add_animation_to_tree)
```

However, it seems I would need to make changes to the `buffers`, `bufferviews` and `accessors` also based on what I read [here](https://github.com/KhronosGroup/glTF-Tutorials/blob/master/gltfTutorial/gltfTutorial_006_SimpleAnimation.md#the-buffer-and-the-bufferview-for-the-raw-animation-data).

That would mean I would need a way for accessing and updating the buffers. So I tried with adding an additional keyword argument  `buffer_postprocessor` that takes in `buffer_items` and `tree` which I think might be part of a way of doing this:

// trimesh.exchange.gltf.py
```python
# allow custom postprocessing
if tree_postprocessor is not None:
    tree_postprocessor(tree)

if buffer_postprocessor is not None:
    buffer_postprocessor(buffer_items, tree)

```
// polygon_animation.py
```python
scene = trimesh.Scene()

scene.add_geometry(polygon_mesh, node_name="test", geom_name="test")

# https://github.com/KhronosGroup/glTF-Tutorials/blob/master/gltfTutorial/gltfTutorial_006_SimpleAnimation.md
# https://github.com/KhronosGroup/glTF-Tutorials/blob/master/gltfTutorial/gltfTutorial_007_Animations.md
def add_animation_to_tree(tree):
    tree["animations"] = [
        {
            "samplers": [{"input": 2, "interpolation": "LINEAR", "output": 3}],
            "channels": [{"sampler": 0, "target": {"node": 1, "path": "rotation"}}],
        }
    ]

def add_animation_to_buffer(buffer_items, tree):
    from trimesh.exchange.gltf import _data_append, uint32

    acc_face = _data_append(
        acc=tree["accessors"],
        buff=buffer_items,
        blob={"componentType": 5125, "type": "SCALAR"},
        data=polygon_mesh.faces.astype(uint32),
    )

scene.export(
    file_obj="polygon_animation.glb",
    file_type=".glb",
    tree_postprocessor=add_animation_to_tree,
    buffer_postprocessor=add_animation_to_buffer,
)
```

This obviously doesn't work straight away as I for 1) haven't looked into what data the animation buffer really expects, and 2) I haven't paid any attention to how the `accessor` is updated. So for now running this code returns an IndexError at `trimesh.exchange.gltf.py:522`

```
        # store the accessor and return the index
>       acc[key] = blob
E       IndexError: cannot fit 'int' into an index-sized integer
```

While I keep trying to figure this out, maybe anyone here might have some insight into the GLTF/trimesh data models that could help me in any way? :) 

Best Regards
Kristoffer

--- a/trimesh/exchange/gltf.py
+++ b/trimesh/exchange/gltf.py
@@ -157,7 +157,8 @@ def export_glb(
         scene,
         include_normals=None,
         unitize_normals=False,
-        tree_postprocessor=None):
+        tree_postprocessor=None,
+        buffer_postprocessor=None):
     """
     Export a scene as a binary GLTF (GLB) file.
 
@@ -187,7 +188,7 @@ def export_glb(
     tree, buffer_items = _create_gltf_structure(
         scene=scene,
         unitize_normals=unitize_normals,
-        include_normals=include_normals)
+        include_normals=include_normals, buffer_postprocessor=buffer_postprocessor)
 
     # allow custom postprocessing
     if tree_postprocessor is not None:
@@ -589,7 +590,8 @@ def _jsonify(blob):
 def _create_gltf_structure(scene,
                            include_normals=None,
                            include_metadata=True,
-                           unitize_normals=None,):
+                           unitize_normals=None,
+                           buffer_postprocessor=None):
     """
     Generate a GLTF header.
 
@@ -682,6 +684,9 @@ def _create_gltf_structure(scene,
     nodes = scene.graph.to_gltf(
         scene=scene, mesh_index=mesh_index)
     tree.update(nodes)
+
+    if buffer_postprocessor is not None:
+        buffer_postprocessor(buffer_items, tree)
 
     # convert accessors back to a flat list
     tree['accessors'] = list(tree['accessors'].values())

